### PR TITLE
refactor(module:*): rework for ivy

### PR DESCRIPTION
**Pull request by hsuanxyz**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/NG-ZORRO/ng-zorro-antd/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[x] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Application (the showcase website) / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: N/A

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: components/slider/nz-slider.component.ts [hunk 1]
```diff
@@ -110,12 +111,15 @@ export class NzSliderComponent implements ControlValueAccessor, OnInit, OnChange
...
     }
     this.toggleDragDisabled(this.nzDisabled);
-
     if (this.getValue() === null) {
       this.setValue(this.formatValue(null));
     }
   }
 
+  ngAfterViewInit(): void {
+    setTimeout(() => {});
```
Comment (wendzhue) [inline components/slider/nz-slider.component.ts]: Is it some hack?

File: components/slider/nz-slider-handle.component.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ export class NzSliderHandleComponent implements OnChanges, OnDestroy {
 
   private updateStyle(): void {
     this.style[this.nzVertical ? 'bottom' : 'left'] = `${this.nzOffset}%`;
+    console.log(this.style);
```
Comment (wendzhue) [inline components/slider/nz-slider-handle.component.ts]:
```diff
- console.log()
```

File: components/layout/nz-sider.component.ts [hunk 3]
```diff
@@ -65,16 +65,22 @@ export class NzSiderComponent implements OnInit, AfterViewInit, OnDestroy {
...
   @Input() nzTheme: 'light' | 'dark' = 'dark';
   @Input() nzCollapsedWidth = 80;
   @Input() nzBreakpoint: NzBreakPoint;
   @Input() nzZeroTrigger: TemplateRef<void>;
-  @Input() @ViewChild('defaultTrigger', { static: true }) nzTrigger: TemplateRef<void>;
   @Input() @InputBoolean() nzReverseArrow = false;
   @Input() @InputBoolean() nzCollapsible = false;
   @Input() @InputBoolean() nzCollapsed = false;
-  @Output() readonly nzCollapsedChange = new EventEmitter();
+  @Input() nzTrigger: TemplateRef<void>;
```
Comment (vthinkxie) [inline components/layout/nz-sider.component.ts]:
```suggestion
  @Input() nzTrigger: TemplateRef<void> = this.defaultTrigger;
```

File: components/pagination/nz-pagination.component.ts [hunk 4]
```diff
@@ -47,10 +52,13 @@ export class NzPaginationComponent implements OnInit, OnDestroy, OnChanges {
   @Input() nzInTable = false;
   @Input() nzSize: 'default' | 'small' = 'default';
   @Input() nzPageSizeOptions = [10, 20, 30, 40];
-  @Input() @ViewChild('renderItemTemplate', { static: true }) nzItemRender: TemplateRef<{
-    $implicit: 'page' | 'prev' | 'next';
-    page: number;
-  }>;
+
+  @Input() nzItemRender: TemplateRef<PaginationItemRenderContext>;
```
Comment (vthinkxie) [inline components/pagination/nz-pagination.component.ts]:
```suggestion
  @Input() nzItemRender: TemplateRef<PaginationItemRenderContext> = this. nzItemRenderChild;
```

File: components/table/nz-table.component.ts [hunk 5]
```diff
@@ -101,10 +102,12 @@ export class NzTableComponent<T = any> implements OnInit, AfterViewInit, OnDestr
   @Input() nzData: T[] = [];
   @Input() nzPaginationPosition: 'top' | 'bottom' | 'both' = 'bottom';
   @Input() nzScroll: { x?: string | null; y?: string | null } = { x: null, y: null };
-  @Input() @ViewChild('renderItemTemplate', { static: true }) nzItemRender: TemplateRef<{
-    $implicit: 'page' | 'prev' | 'next';
-    page: number;
-  }>;
+
+  @Input() nzItemRender: TemplateRef<PaginationItemRenderContext>;
```
Comment (vthinkxie) [inline components/table/nz-table.component.ts]:
```suggestion
  @Input() nzItemRender: TemplateRef<PaginationItemRenderContext> = this.itemRenderChild;
```

File: components/tree-select/nz-tree-select.component.ts [hunk 6]
```diff
@@ -153,7 +153,11 @@ export class NzTreeSelectComponent extends NzTreeBase implements ControlValueAcc
   @ViewChild(CdkOverlayOrigin, { static: true }) cdkOverlayOrigin: CdkOverlayOrigin;
   @ViewChild(CdkConnectedOverlay, { static: false }) cdkConnectedOverlay: CdkConnectedOverlay;
 
-  @Input() @ContentChild('nzTreeTemplate', { static: true }) nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }>;
+  @Input() nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }>;
```
Comment (vthinkxie) [inline components/tree-select/nz-tree-select.component.ts]:
```suggestion
  @Input() nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }> = this. nzTreeTemplateChild;
```

File: components/tree/nz-tree.component.ts [hunk 7]
```diff
@@ -83,11 +83,14 @@ export class NzTreeComponent extends NzTreeBase implements OnInit, OnDestroy, Co
   @Input() @InputBoolean() nzSelectMode = false;
   @Input() @InputBoolean() nzCheckStrictly = false;
   @Input() @InputBoolean() nzBlockNode = false;
-
-  @Input() @ContentChild('nzTreeTemplate', { static: true }) nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }>;
-
   @Input() @InputBoolean() nzExpandAll = false;
 
+  @Input() nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }>;
```
Comment (vthinkxie) [inline components/tree/nz-tree.component.ts]:
```suggestion
  @Input() nzTreeTemplate: TemplateRef<{ $implicit: NzTreeNode }> = this.nzTreeTemplateChild;
```

**Pull request merged**